src/components/sdk/MeetingControls/AudioInputVFControl.tsx

```diff
@@ -0,0 +1,151 @@
+import { useAudioInputs } from '../../../providers/DevicesProvider';
+import { useVoiceFocus } from '../../../providers/VoiceFocusProvider';
+import { useToggleLocalMute } from '../../../hooks/sdk/useToggleLocalMute';
+import { DeviceConfig } from '../../../types';
+import { isOptionActive, audioInputSelectionToDevice } from '../../../utils/device-utils';
+import { PopOverItemProps } from '../../ui/PopOver/PopOverItem';
+import { PopOverSeparator } from '../../..';
+import { Spinner } from '../../ui/icons';
+
+interface Props {
+  /** The label that will be shown when microphone is muted , it defaults to `Mute`. */
+  muteLabel?: string;
+  /** The label that will be shown when microphone is unmuted, it defaults to `Unmute`. */
+  unmuteLabel?: string;
+  /** Title attribute for the icon when muted, it defaults to `Muted microphone` */
+  mutedIconTitle?: string;
+  /** Title attribute for the icon when unmuted, it defaults to `Microphone` */
+  unmutedIconTitle?: string;
+}
+
+const AudioInputVFControl: React.FC<Props> = ({
+  muteLabel = 'Mute',
+  unmuteLabel = 'Unmute',
+  mutedIconTitle,
+  unmutedIconTitle
+}) => {
+  const meetingManager = useMeetingManager();
+  const [ voiceFocusOn, setVoiceFocusOn ] = useState(false);
+  const [ isLoading, setIsloading ] = useState(false);
+  const [ enableVoiceFocus, setEnableVoiceFocus] = useState(false);
+  const { muted, toggleMute } = useToggleLocalMute();
+  const [ device, setDevice ] = useState<Device | AudioTransformDevice | null>(meetingManager.selectedAudioInputTransformDevice);
+  const { supportsVoiceFocus, transformToVoiceFocus } = useVoiceFocus();
+  const audioInputConfig: DeviceConfig = {
+    additionalDevices: true,
+  };
+  const { devices, selectedDevice } = useAudioInputs(audioInputConfig);
+
+  useEffect(() => {
+    if (device instanceof VoiceFocusTransformDevice) {
+        setEnableVoiceFocus(true);
+    } else {
+        setEnableVoiceFocus(false);
+    }
+  }, [device]);
+
+  useEffect(() => {
+    const callback = (updatedAudioInputDevice: Device | AudioTransformDevice | null): void => {
+        setDevice(updatedAudioInputDevice);
+    };
+    meetingManager.subscribeToSelectedAudioInputTransformDevice(callback);
+
+    return (): void => {
+      meetingManager.unsubscribeFromSelectedAudioInputTransformDevice(callback);
+    };
+  }, []);
+
+  useEffect(() => {
+    if (supportsVoiceFocus) {
+        dropdownOptions.push({
+            children: <><PopOverSeparator />{isLoading && <Spinner />} {enableVoiceFocus ? <span>Disable voice focus</span> : <span>Enable voice focus</span>}</>,
+            checked: enableVoiceFocus,
+            disabled: isLoading,
+            onClick: () => {
+                setVoiceFocusOn(current => !current);
+              },
+        });
+    }
+  }, [supportsVoiceFocus, enableVoiceFocus, voiceFocusOn, isLoading]);
+
+  useEffect(() => {
+    console.log(`now the voice focus is supported: ${supportsVoiceFocus}` );
```

Comment:
Spurious logging.

Reply:
Sure, removed this block of code. This is for testing, but I forgot to remove it.